dts/bindings/sensor/hamamatsu,s11059.yaml

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2022, Hiroki Tada
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Hamamatsu Photonics S11059 Color Sensor.  See datasheet at
+  https://datasheetspdf.com/pdf/1323325/Hamamatsu/S11059-02DT/1
+
+compatible: "hamamatsu,s11059"
+
+include: i2c-device.yaml
+
+properties:
+
+  gain:
+    type: int
```

Comment:
Why not a boolean?

Reply:
This value is used to select high-gain or low-gain.
So, I thought it would be appropriate to use Enum instead of Boolean.

Reply:
Will there ever be a third option?

Reply:
Changed property name from "gain" to "high-gain" to use boolean.